binding/form_mapping.go

```diff
@@ -197,6 +197,18 @@ func setByForm(value reflect.Value, field reflect.StructField, form map[string][
 	}
 }
 
+type jsonUnmarshaler interface {
+	UnmarshalJSON([]byte) error
```

Comment:
What's the reason to use multiple different interfaces? Given that these will always execute in a specific order. `JSON` then `TEXT` then `BINARY`. If all 3 are defined only `JSON` one will be executed. So in this case only 1 interface would be enough to define the unmarshaller override behaviour. E.g. `func Unmarshal([]byte) error`